Source/WebCore/platform/VideoPixelFormat.cpp

```diff
@@ -43,6 +47,34 @@ VideoPixelFormat convertVideoFramePixelFormat(uint32_t format, bool shouldDiscar
     if (format == kCVPixelFormatType_32ARGB)
         return shouldDiscardAlpha ? VideoPixelFormat::RGBX : VideoPixelFormat::RGBA;
     ASSERT_NOT_REACHED();
+#elif USE(GSTREAMER)
+    switch (format) {
+    case GST_VIDEO_FORMAT_I420:
+        return VideoPixelFormat::I420;
+    case GST_VIDEO_FORMAT_A420:
+        return VideoPixelFormat::I420A;
+    case GST_VIDEO_FORMAT_I422_10BE:
+    case GST_VIDEO_FORMAT_I422_10LE:
+    case GST_VIDEO_FORMAT_I422_12BE:
+    case GST_VIDEO_FORMAT_I422_12LE:
+        return VideoPixelFormat::I422;
+    case GST_VIDEO_FORMAT_Y444:
+        return VideoPixelFormat::I444;
+    case GST_VIDEO_FORMAT_NV12:
+        return VideoPixelFormat::NV12;
+    case GST_VIDEO_FORMAT_RGBA:
+        return shouldDiscardAlpha ? VideoPixelFormat::RGBX : VideoPixelFormat::RGBA;
+    case GST_VIDEO_FORMAT_RGBx:
+        return VideoPixelFormat::RGBX;
+    case GST_VIDEO_FORMAT_BGRA:
+        return shouldDiscardAlpha ? VideoPixelFormat::BGRX : VideoPixelFormat::BGRA;
+    case GST_VIDEO_FORMAT_ARGB:
+        return shouldDiscardAlpha ? VideoPixelFormat::RGBX : VideoPixelFormat::RGBA;
```

Comment:
Is this correct or a copy&paste error?

Reply:
looks fishy indeed

Reply:
Ah I think I did the same as on the Cocoa code path... But I suspect we'd instead need a formal `ARGB` in `VideoPixelFormat`. WDYT @youennf ?

Reply:
`VideoPixelFormat` is bound to WebIDL defined https://www.w3.org/TR/webcodecs/#enumdef-videopixelformat, which does not have ARGB.
What we have done in Cocoa is to relate WebCodecs RGBA to CVPixelBuffer ARGB, see Cocoa VideoFrame::createRGBA.

Reply:
I suppose we'll have to do the same then :)